### ISSUE TITLE
Skip batch when OOM

### ISSUE THREAD

**Issue by candygocandy** [huggingface/trl#1189]
Hello Folks

Great work on the DPO. I got a question, my script was running fine with the DPO training, but at some batch, it is causing OOM. I can't reduce the batch size between it is already 2. How do I skip batch that cause the OOM? Or is there a work around method on it?

My guess is that either the example is too long, or the generation is too long. Currently, I cap the max_seq size = 800, but I don't know what is the best number. 

Please help. Thanks

**Comment by lvwerra** [comment]
There are a few options to reduce memory consumption:
1. Use BS=1 (and scale gradient accumulation)
2. You could use PEFT to reduce the memory footprint
3. Flash Attention 2 should also reduce the memory footprint
4. Make sure you have gradient ckeckpointing activated.
